M2/Macaulay2/packages/KGAlgebras.m2

```diff
@@ -3,132 +3,115 @@ newPackage(
     Version => "0.5",
     Date => "February 19th, 2025",
     Authors => {
-	{Name => "Daniel Koizumi", Email => "daniel.koizumi@utexas.edu"},
+    {Name => "Daniel Koizumi", Email => "daniel.koizumi@utexas.edu"},
     {Name => "Moses Samuelson-Lynn", Email => "m.samuelsonlynn@utah.edu"},
     {Name => "Toshi Mowery", Email => "tosmow@gmail.com"}
     },
-    Headline => "A package for KG-algebras",
-    Keywords => {"KG-algebra"},
-    DebuggingMode => false,
-    Reload=>true,
+    Headline => "A package for group algebras",
+    Keywords => {"Group Theory"},
+    DebuggingMode => true,
     PackageExports=>{"AssociativeAlgebras","InvariantRing","RealRoots"}
     )
-export {"KGAlg","meataxe", "SnGroup"}
+export {"GroupRing","groupRing","meataxe", "SnGroup"}
 
-KGAlgebra = new Type of FreeAlgebraQuotient;
-KGAlg = method()
-KGAlg (FiniteGroupAction, Ring) := (G,K) -> (
-	if not isField K then (
-		error "KGModules: Expected second argument to be a field.";
-		);
-	matrixElts := group G; -- create set of matrix elements
-	n := #matrixElts-1; -- maximum index, since we are starting at 0
-	groupIndices := {0..n}; -- indices
-	g := symbol g;
-	groupElts := g_0 .. g_n; -- group elements
-	multTableIdealGens := {};
-	FA := freeAlgebra(K,groupElts);
-	use FA; -- make FA our working ring
-	local H;
-	local productInv;
-	local multTableIdealGens;
-	for i in 0 .. n do(
-		for j in 0 .. n do( 
-			productInv = position(matrixElts,M -> M==matrixElts#i * matrixElts#j);
-			H= (FA_i)*(FA_j)-(FA_productInv);
-			multTableIdealGens=append(multTableIdealGens,H);
-		);
-	);
-	FA/ideal(multTableIdealGens) -- output
+GroupRing = new Type of FreeAlgebraQuotient;
+groupRing = method()
+groupRing (FiniteGroupAction, Ring) := (G,K) -> (
+    if not isField K then (
+        error "KGModules: Expected second argument to be a field.";
+    );
+    matrixElts := group G; -- create set of matrix elements
+    n := #matrixElts-1; -- maximum index, since we are starting at 0
+    groupIndices := {0..n}; -- indices
+    g := symbol g;
+    groupElts := g_0 .. g_n; -- group elements
+    FA := freeAlgebra(K,groupElts);
+    use FA; -- make FA our working ring
+    multTableIdealGens := new MutableList;
+    for i in 0 .. n do(
+        for j in 0 .. n do( 
+            productInv := position(matrixElts,M -> M==matrixElts#i * matrixElts#j);
+            H := (FA_i)*(FA_j)-(FA_productInv);
+            multTableIdealGens##multTableIdealGens = H;
+        );
+    );
+    print multTableIdealGens;
```

Comment:
Did you mean to print this?  I'd either remove it or swap it with something like `if debugLevel > something then print ...`.